--- snownews-1.5.10.original/opml2snow	2008-07-28 11:26:16.000000000 +0200
+++ snownews-1.5.10/opml2snow	2008-09-04 12:02:35.000000000 +0200
@@ -29,6 +29,16 @@
 use English;
 
 #######################################################################
+# etc_path
+    my($ETC_PATH);    
+
+    if ($ENV{'HOME_ETC'}) {
+    $ETC_PATH = ("$ENV{'HOME_ETC'}");
+    } else {
+    $ETC_PATH = ($ENV{'HOME'});
+    }
+
+#######################################################################
 # Help
 if (index($ARGV[0], "-h") >= 0) {
 	print "Snownews opml2snow - subsription file import/export utility\n\n".
@@ -80,7 +90,7 @@
 	if (defined($ARGV[1])) {
 		$inputfile = $ARGV[1];
 	} else {
-		$inputfile = "$ENV{'HOME'}/.snownews/urls"
+		$inputfile = "$ETC_PATH/.snownews/urls"
 	}
 	open (URLS, "$inputfile");
 	my(@urls) = <URLS>;
@@ -123,7 +133,7 @@
 		my($feed_script) = 4;
 		
 		my($parser) = XML::LibXML->new();
-		my($feed_doc) = $parser->parse_file("$ENV{'HOME'}/.snownews/cache/$feed_url_hash");
+		my($feed_doc) = $parser->parse_file("$ETC_PATH/.snownews/cache/$feed_url_hash");
 		my($feed_root) = $feed_doc->documentElement();
 		$feed_root->setNamespace('http://purl.org/rss/1.0/', 'rss', 1);
 		
