--- snownews-1.5.10/Makefile	2008-09-01 16:50:37.000000000 +0200
+++ snownews-1.5.10.b/Makefile	2008-09-01 16:44:24.000000000 +0200
@@ -3,6 +3,7 @@
 INSTALL=    install
 LOCALEPATH= $(PREFIX)/share/locale
 MANPATH=    $(PREFIX)/share/man
+BINDIR=     $(PREFIX)/bin
 
 ### Compiler/linker flags   ###
 ### Generated via configure ###
@@ -57,13 +58,13 @@
 	@echo ""
 
 install-bin: snownews
-	if [ ! -d "$(DESTDIR)$(PREFIX)/bin" ]; then \
-		mkdir -p $(DESTDIR)$(PREFIX)/bin; \
+	if [ ! -d "$(DESTDIR)$(BINDIR)" ]; then \
+		mkdir -p $(DESTDIR)$(BINDIR); \
 	fi
-	$(INSTALL) -s snownews $(DESTDIR)$(PREFIX)/bin
-	$(INSTALL) opml2snow $(DESTDIR)$(PREFIX)/bin
-	if [ ! -f "$(DESTDIR)$(PREFIX)/bin/snow2opml" ]; then \
-		(cd $(DESTDIR)$(PREFIX)/bin && \
+	$(INSTALL) -s snownews $(DESTDIR)$(BINDIR)
+	$(INSTALL) opml2snow $(DESTDIR)$(BINDIR)
+	if [ ! -f "$(DESTDIR)$(BINDIR)/snow2opml" ]; then \
+		(cd $(DESTDIR)$(BINDIR) && \
 		 ln -sf opml2snow snow2opml ); \
 	fi;
 
