diff -Naur snownews-1.5.10.original/Makefile snownews-1.5.10/Makefile
--- snownews-1.5.10.original/Makefile	2008-09-04 11:40:58.000000000 +0200
+++ snownews-1.5.10/Makefile	2008-09-04 11:42:12.000000000 +0200
@@ -10,15 +10,15 @@
 include platform_settings
 
 ### Object files ###
-OBJFILES= main.o netio.o interface.o xmlparse.o updatecheck.o conversions.o dialog.o ui-support.o categories.o about.o cookies.o setup.o net-support.o digcalc.o md5.o filters.o io-internal.o os-support.o zlib_interface.o support.o
+OBJFILES= main.o netio.o interface.o xmlparse.o updatecheck.o conversions.o dialog.o ui-support.o categories.o about.o cookies.o setup.o net-support.o digcalc.o md5.o filters.o io-internal.o os-support.o zlib_interface.o support.o etc_path.o
 
 VERSION= `grep VERSION version.h | sed s/\"//g | sed s/\#define\ VERSION\ //`
 DISTDIR= snownews-$(VERSION)
 DISTFILES = AUTHOR COPYING CREDITS Changelog README README.de README.patching INSTALL opml2snow \
 	Makefile configure \
 	doc po scripts \
-	main.c interface.c netio.c xmlparse.c updatecheck.c os-support.c conversions.c dialog.c ui-support.c categories.c about.c cookies.c setup.c net-support.c digcalc.c md5.c filters.c io-internal.c zlib_interface.c support.c \
-	config.h version.h main.h interface.h netio.h xmlparse.h updatecheck.h os-support.h conversions.h dialog.h ui-support.h categories.h about.h cookies.h setup.h net-support.h digcalc.h md5.h filters.h io-internal.h zlib_interface.h support.h
+	main.c interface.c netio.c xmlparse.c updatecheck.c os-support.c conversions.c dialog.c ui-support.c categories.c about.c cookies.c setup.c net-support.c digcalc.c md5.c filters.c io-internal.c zlib_interface.c support.c etc_path.c\
+	config.h version.h main.h interface.h netio.h xmlparse.h updatecheck.h os-support.h conversions.h dialog.h ui-support.h categories.h about.h cookies.h setup.h net-support.h digcalc.h md5.h filters.h io-internal.h zlib_interface.h support.h etc_path.h
 
 ### Translations ###
 LOCALES= de es fr it nl ru sl se zh_TW zh_CN pt_BR pl ja be@latin uk_UA
diff -Naur snownews-1.5.10.original/cookies.c snownews-1.5.10/cookies.c
--- snownews-1.5.10.original/cookies.c	2008-07-28 11:26:16.000000000 +0200
+++ snownews-1.5.10/cookies.c	2008-09-04 11:42:12.000000000 +0200
@@ -34,6 +34,7 @@
 
 #include "config.h"
 #include "ui-support.h"
+#include "etc_path.h"
 
 void CookieCutter (struct feed * cur_ptr, FILE * cookies) {
 	char buf[4096];					/* File read buffer. */
@@ -209,7 +210,7 @@
 	char file[512];					/* File locations. */
 	FILE *cookies;					/* Cookies file ptr. */
 	
-	snprintf (file, sizeof(file), "%s/.snownews/cookies", getenv("HOME"));
+	snprintf (file, sizeof(file), "%s/.snownews/cookies", set_config_dir());
 	cookies = fopen (file, "r");
 	if (cookies == NULL) {
 		/* No cookies to load. */
diff -Naur snownews-1.5.10.original/dialog.c snownews-1.5.10/dialog.c
--- snownews-1.5.10.original/dialog.c	2008-07-28 11:26:16.000000000 +0200
+++ snownews-1.5.10/dialog.c	2008-09-04 11:42:12.000000000 +0200
@@ -39,6 +39,7 @@
 #include "dialog.h"
 #include "io-internal.h"
 #include "conversions.h"
+#include "etc_path.h"
 
 extern char *browser;
 extern struct keybindings keybindings;
@@ -293,9 +294,9 @@
 	centerx = COLS / 2;
 	
 	hashme = Hashify(current_feed->feedurl);
-	len = (strlen(getenv("HOME")) + strlen(hashme) + 18);
+	len = (strlen(set_config_dir()) + strlen(hashme) + 18);
 	file = malloc (len);
-	snprintf (file, len, "%s/.snownews/cache/%s", getenv("HOME"), hashme);
+	snprintf (file, len, "%s/.snownews/cache/%s", set_config_dir(), hashme);
 
 	UISupportDrawBox (5, 4, COLS-6, 13);
 	
diff -Naur snownews-1.5.10.original/etc_path.c snownews-1.5.10/etc_path.c
--- snownews-1.5.10.original/etc_path.c	1970-01-01 01:00:00.000000000 +0100
+++ snownews-1.5.10/etc_path.c	2008-09-04 11:42:12.000000000 +0200
@@ -0,0 +1,12 @@
+/*
+ * Set path for preferences. <l00natyk@gazeta.pl>
+ */
+
+#include <stdlib.h>
+
+char *set_config_dir(void) {
+	char *etc_dir;
+	if ((etc_dir = getenv("HOME_ETC")) != NULL) return etc_dir;
+	etc_dir = getenv("HOME");
+	return etc_dir;
+	}
diff -Naur snownews-1.5.10.original/etc_path.h snownews-1.5.10/etc_path.h
--- snownews-1.5.10.original/etc_path.h	1970-01-01 01:00:00.000000000 +0100
+++ snownews-1.5.10/etc_path.h	2008-09-04 11:42:12.000000000 +0200
@@ -0,0 +1,6 @@
+#ifndef ETC_PATH_H
+#define	ETC_PATH_H
+
+char *set_config_dir (void);
+
+#endif
diff -Naur snownews-1.5.10.original/interface.c snownews-1.5.10/interface.c
--- snownews-1.5.10.original/interface.c	2008-07-28 11:26:16.000000000 +0200
+++ snownews-1.5.10/interface.c	2008-09-04 11:42:12.000000000 +0200
@@ -46,6 +46,7 @@
 #include "io-internal.h"
 #include "os-support.h"
 #include "support.h"
+#include "etc_path.h"
 
 extern struct keybindings keybindings;
 extern struct color color;
@@ -1323,9 +1324,9 @@
 						
 						/* Remove cachefile from filesystem. */
 						hashme = Hashify(highlighted->feedurl);
-						len = (strlen(getenv("HOME")) + strlen(hashme) + 18);
+						len = (strlen(set_config_dir()) + strlen(hashme) + 18);
 						file = malloc (len);
-						snprintf (file, len, "%s/.snownews/cache/%s", getenv("HOME"), hashme);
+						snprintf (file, len, "%s/.snownews/cache/%s", set_config_dir(), hashme);
 						
 						/* Errors from unlink can be ignored. Worst thing that happens is that
 						   we delete a file that doesn't exist. */
diff -Naur snownews-1.5.10.original/io-internal.c snownews-1.5.10/io-internal.c
--- snownews-1.5.10.original/io-internal.c	2008-07-28 11:26:16.000000000 +0200
+++ snownews-1.5.10/io-internal.c	2008-09-04 11:42:12.000000000 +0200
@@ -43,6 +43,7 @@
 #include "xmlparse.h"
 #include "ui-support.h"
 #include "io-internal.h"
+#include "etc_path.h"
 
 extern char *browser;
 
@@ -276,7 +277,7 @@
 		return 0;
 	
 	hashme = Hashify(cur_ptr->feedurl);
-	snprintf (file, sizeof(file), "%s/.snownews/cache/%s", getenv("HOME"), hashme);
+	snprintf (file, sizeof(file), "%s/.snownews/cache/%s", set_config_dir(), hashme);
 	free (hashme);
 	cache = fopen (file, "r");
 	
@@ -388,7 +389,7 @@
 	UIStatus (_("Saving settings ["), 0, 0);
 	titlestrlen = strlen (_("Saving settings ["));
 	
-	snprintf (file, sizeof(file), "%s/.snownews/browser", getenv("HOME"));
+	snprintf (file, sizeof(file), "%s/.snownews/browser", set_config_dir());
 	configfile = fopen (file, "w+");
 	if (configfile == NULL) {
 		MainQuit (_("Save settings (browser)"), strerror(errno));
@@ -396,12 +397,12 @@
 	fputs (browser, configfile);
 	fclose (configfile);
 	
-	snprintf (file, sizeof(file), "%s/.snownews/urls", getenv("HOME"));
+	snprintf (file, sizeof(file), "%s/.snownews/urls", set_config_dir());
 	
 	/* Make a backup of urls. This approach is really broken! */
 	if ((stat (file, &filetest)) != -1) {
 		if ((filetest.st_mode & S_IFREG) == S_IFREG) {
-			snprintf (syscall, sizeof(file), "cp -f %s/.snownews/urls %s/.snownews/urls.bak", getenv("HOME"), getenv("HOME"));
+			snprintf (syscall, sizeof(file), "cp -f %s/.snownews/urls %s/.snownews/urls.bak", set_config_dir(), set_config_dir());
 			system (syscall);
 		}
 	}
@@ -460,7 +461,7 @@
 		 * Write cache.
 		 */
 		hashme = Hashify(cur_ptr->feedurl);
-		snprintf (file, sizeof(file), "%s/.snownews/cache/%s", getenv("HOME"), hashme);
+		snprintf (file, sizeof(file), "%s/.snownews/cache/%s", set_config_dir(), hashme);
 		free (hashme);
 		cache = fopen (file, "w+");
 
diff -Naur snownews-1.5.10.original/setup.c snownews-1.5.10/setup.c
--- snownews-1.5.10.original/setup.c	2008-07-28 11:26:16.000000000 +0200
+++ snownews-1.5.10/setup.c	2008-09-04 11:42:12.000000000 +0200
@@ -40,6 +40,7 @@
 #include "main.h"
 #include "setup.h"
 #include "io-internal.h"
+#include "etc_path.h"
 
 struct feed *first_ptr = NULL;
 struct entity *first_entity = NULL;
@@ -559,7 +560,7 @@
 	/* 
 	 * Setup config directories.
 	 */
-	snprintf (file, sizeof(file), "%s/.snownews", getenv("HOME"));
+	snprintf (file, sizeof(file), "%s/.snownews", set_config_dir());
 	if ((stat (file, &dirtest)) == -1 ) {
 		/* Create directory. */
 		if (mkdir (file, S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH) != 0)
@@ -572,7 +573,7 @@
 		}
 	}
 	
-	snprintf (file, sizeof(file), "%s/.snownews/cache", getenv("HOME"));
+	snprintf (file, sizeof(file), "%s/.snownews/cache", set_config_dir());
 	if ((stat (file, &dirtest)) == -1) {
 		/* Create directory. */
 		if (mkdir (file, S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH) != 0)
@@ -586,20 +587,20 @@
 	
 	/* Redirect stderr to ~/.snownews/error.log
 	   Be sure to call _after_ the directory checks above! */
-	snprintf (file, sizeof(file), "%s/.snownews/error.log", getenv("HOME"));
+	snprintf (file, sizeof(file), "%s/.snownews/error.log", set_config_dir());
 	errorlog = fopen (file, "w+");
 	dup2 (fileno(errorlog), STDERR_FILENO);
 	
 	UIStatus (_("Reading configuration settings..."), 0, 0);
 	
-	snprintf (file, sizeof(file), "%s/.snownews/browser", getenv("HOME"));
+	snprintf (file, sizeof(file), "%s/.snownews/browser", set_config_dir());
 	SetupBrowser (file);
 
 	/*************
 	 * Feed list *
 	 *************/
 	
-	snprintf (file, sizeof(file), "%s/.snownews/urls", getenv("HOME"));
+	snprintf (file, sizeof(file), "%s/.snownews/urls", set_config_dir());
 	configfile = fopen (file, "r");
 	if (configfile == NULL) {
 		UIStatus (_("Creating new configfile."), 0, 0);
@@ -694,13 +695,13 @@
 	}
 	fclose (configfile);
 	
-	snprintf (file, sizeof(file), "%s/.snownews/keybindings", getenv("HOME"));
+	snprintf (file, sizeof(file), "%s/.snownews/keybindings", set_config_dir());
 	SetupKeybindings (file);
 	
-	snprintf (file, sizeof(file), "%s/.snownews/colors", getenv("HOME"));
+	snprintf (file, sizeof(file), "%s/.snownews/colors", set_config_dir());
 	SetupColors (file);
 	
-	snprintf (file, sizeof(file), "%s/.snownews/html_entities", getenv("HOME"));
+	snprintf (file, sizeof(file), "%s/.snownews/html_entities", set_config_dir());
 	SetupEntities (file);
 	
 	return numfeeds;
diff -Naur snownews-1.5.10.original/support.c snownews-1.5.10/support.c
--- snownews-1.5.10.original/support.c	2008-07-28 11:26:16.000000000 +0200
+++ snownews-1.5.10/support.c	2008-09-04 11:42:12.000000000 +0200
@@ -28,6 +28,7 @@
 #include <sys/stat.h>
 #include <sys/select.h>
 #include <time.h>
+#include "etc_path.h"
 
 int easterEgg (void) {
 	struct tm *t;
@@ -39,7 +40,7 @@
 	tunix = time(0);
 	t = localtime(&tunix);
 	
-	snprintf (file, sizeof(file), "%s/.snownews/santa_hunta", getenv("HOME"));
+	snprintf (file, sizeof(file), "%s/.snownews/santa_hunta", set_config_dir());
 	if ((stat (file, &filetest) == 0) ||
 		((t->tm_mon == 11) && (t->tm_mday >= 24) && (t->tm_mday <= 26))) {
 		return 1;
diff -Naur snownews-1.5.10.original/ui-support.c snownews-1.5.10/ui-support.c
--- snownews-1.5.10.original/ui-support.c	2008-07-28 11:26:16.000000000 +0200
+++ snownews-1.5.10/ui-support.c	2008-09-04 11:42:12.000000000 +0200
@@ -32,6 +32,7 @@
 #include "version.h"
 #include "ui-support.h"
 #include "io-internal.h"
+#include "etc_path.h"
 
 extern struct color color;
 extern char *browser;
@@ -392,7 +393,7 @@
 	else
 		pager = strdup("less");
 	
-	env = getenv("HOME");
+	env = set_config_dir();
 	snprintf (errorlog, sizeof(errorlog), "%s/.snownews/error.log", env);
 	
 	snprintf (command, sizeof(command), "%s %s", pager, errorlog);
diff -Naur snownews-1.5.10.original/updatecheck.c snownews-1.5.10/updatecheck.c
--- snownews-1.5.10.original/updatecheck.c	2008-07-28 11:26:16.000000000 +0200
+++ snownews-1.5.10/updatecheck.c	2008-09-04 11:42:12.000000000 +0200
@@ -34,7 +34,7 @@
 #include "version.h"
 #include "netio.h"
 #include "io-internal.h"
-
+#include "etc_path.h"
 
 void AutoVersionCheck (void) {
 	struct feed *update;
@@ -51,7 +51,7 @@
 	
 	/* We check once a week. */
 	snprintf (timestring, sizeof(timestring), "%d", (int) time(NULL));
-	snprintf (file, sizeof(file), "%s/.snownews/updatecheck", getenv("HOME"));
+	snprintf (file, sizeof(file), "%s/.snownews/updatecheck", set_config_dir());
 	lastupdated = fopen (file, "r+");
 	if (lastupdated == NULL) {
 		lastupdated = fopen (file, "w+");
