diff -urN snownews-1.5.5.1.orig/Makefile snownews-1.5.5.1/Makefile
--- snownews-1.5.5.1.orig/Makefile	2004-11-23 13:45:17.258462000 +0100
+++ snownews-1.5.5.1/Makefile	2004-11-23 13:48:22.230342128 +0100
@@ -72,17 +72,17 @@
 	done
 
 install-man: manpages
-	if [ ! -d "$(DESTDIR)$(PREFIX)/man/man1" ]; then \
-		mkdir -p $(DESTDIR)$(PREFIX)/man/man1; \
+	if [ ! -d "$(DESTDIR)$(PREFIX)/share/man/man1" ]; then \
+		mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1; \
 	fi
-	$(INSTALL) -m 0644 $(MAN)/snownews.1 $(DESTDIR)$(PREFIX)/man/man1
-	$(INSTALL) -m 0644 $(MAN)/opml2snow.1 $(DESTDIR)$(PREFIX)/man/man1
+	$(INSTALL) -m 0644 $(MAN)/snownews.1 $(DESTDIR)$(PREFIX)/share/man/man1
+	$(INSTALL) -m 0644 $(MAN)/opml2snow.1 $(DESTDIR)$(PREFIX)/share/man/man1
 	
 	for L in $(LANGS); do \
-		if [ ! -d "$(DESTDIR)$(PREFIX)/man/$$L/man1" ]; then \
-			mkdir -p $(DESTDIR)$(PREFIX)/man/$$L/man1; \
+		if [ ! -d "$(DESTDIR)$(PREFIX)/share/man/$$L/man1" ]; then \
+			mkdir -p $(DESTDIR)$(PREFIX)/share/man/$$L/man1; \
 		fi; \
-		$(INSTALL) -m 0644 $(MAN)/$$L/snownews.1 $(DESTDIR)$(PREFIX)/man/$$L/man1; \
+		$(INSTALL) -m 0644 $(MAN)/$$L/snownews.1 $(DESTDIR)$(PREFIX)/share/man/$$L/man1; \
 	done
 
 ### Cleanup ###
