--- snownews-1.5.7.original/Makefile	2005-06-05 11:50:22.000000000 +0200
+++ snownews-1.5.7/Makefile	2005-09-03 20:06:30.000000000 +0200
@@ -2,6 +2,8 @@
 MAKE=       make
 INSTALL=    install
 LOCALEPATH= $(PREFIX)/share/locale
+BINDIR=     $(PREFIX)/bin
+MANDIR=     $(PREFIX)/share/man
 
 ### Compiler/linker flags   ###
 ### Generated via configure ###
@@ -56,15 +58,15 @@
 	@echo ""
 
 install-bin: snownews
-	if [ ! -d "$(DESTDIR)$(PREFIX)/bin" ]; then \
-		mkdir -p $(DESTDIR)$(PREFIX)/bin; \
+	if [ ! -d "$(DESTDIR)$(BINDIR)" ]; then \
+		mkdir -p $(DESTDIR)$(BINDIR); \
 	fi
-	$(INSTALL) -s snownews $(DESTDIR)$(PREFIX)/bin
-	$(INSTALL) opml2snow $(DESTDIR)$(PREFIX)/bin
-	if [ ! -f "$(DESTDIR)$(PREFIX)/bin/snow2opml" ]; then \
-		ln -sf $(DESTDIR)$(PREFIX)/bin/opml2snow $(DESTDIR)$(PREFIX)/bin/snow2opml; \
+	$(INSTALL) snownews $(DESTDIR)$(BINDIR)
+	$(INSTALL) opml2snow $(DESTDIR)$(BINDIR)
+	if [ ! -f "$(DESTDIR)$(BINDIR)/snow2opml" ]; then \
+		ln -sf opml2snow $(DESTDIR)$(BINDIR)/snow2opml; \
 	fi;
-	$(INSTALL) snowsync $(DESTDIR)$(PREFIX)/bin
+	$(INSTALL) snowsync $(DESTDIR)$(BINDIR)
 
 install-locales: locales
 	for L in $(LOCALES); do \
@@ -75,17 +77,17 @@
 	done
 
 install-man: manpages
-	if [ ! -d "$(DESTDIR)$(PREFIX)/man/man1" ]; then \
-		mkdir -p $(DESTDIR)$(PREFIX)/man/man1; \
+	if [ ! -d "$(DESTDIR)$(MANDIR)/man1" ]; then \
+		mkdir -p $(DESTDIR)$(MANDIR)/man1; \
 	fi
-	$(INSTALL) -m 0644 $(MAN)/snownews.1 $(DESTDIR)$(PREFIX)/man/man1
-	$(INSTALL) -m 0644 $(MAN)/opml2snow.1 $(DESTDIR)$(PREFIX)/man/man1
+	$(INSTALL) -m 0644 $(MAN)/snownews.1 $(DESTDIR)$(MANDIR)/man1
+	$(INSTALL) -m 0644 $(MAN)/opml2snow.1 $(DESTDIR)$(MANDIR)/man1
 	
 	for L in $(LANGS); do \
-		if [ ! -d "$(DESTDIR)$(PREFIX)/man/$$L/man1" ]; then \
-			mkdir -p $(DESTDIR)$(PREFIX)/man/$$L/man1; \
+		if [ ! -d "$(DESTDIR)$(MANDIR)/$$L/man1" ]; then \
+			mkdir -p $(DESTDIR)$(MANDIR)/$$L/man1; \
 		fi; \
-		$(INSTALL) -m 0644 $(MAN)/$$L/snownews.1 $(DESTDIR)$(PREFIX)/man/$$L/man1; \
+		$(INSTALL) -m 0644 $(MAN)/$$L/snownews.1 $(DESTDIR)$(MANDIR)/$$L/man1; \
 	done
 
 ### Cleanup ###
